Dockerfile

```diff
@@ -31,13 +31,35 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update -y && \
       ca-certificates \
       curl
 
+# Conda install. 
+#
+# Everything is installed in the root environment. This allows for 
+# upgrades to the packages and eliminates the pain of trying to activate 
+# some other environment automatically for the many different ways 
+# people can use a docker image.
+#
+# Things are pinned to prevent upgrades from conda and force it to 
+# resolve dependencies relative to a fixed conda & python version.
+#
+# Note that the python version is also listed in the enviornment.yml 
+# file. The version in CIVIS_PYTHON_VERSION should take precedence. 
+# If you want to change the python version, you need to change it in 
+# both places. (It could be removed from the environment.yml 
+# file in fact.)
```

Comment:
I have a preference for a single source of truth. Can we axe the one in the `environment.yml`?